resources/linux/debian/postinst.template

```diff
@@ -42,9 +42,15 @@ if [ "@@NAME@@" != "code-oss" ]; then
 		db_get @@NAME@@/add-microsoft-repo || true
 	fi
 
-	# Determine whether to install the repository source list
+	# Determine whether to write the Microsoft repository source list
+	# 0 means do not write
+	# 1 means ask and then write if Yes
+	# 2 means write without asking
```

Comment:
So this would make the default behavior 0 where we no longer write? I think this would break our update story we've had forever and users would get stuck on older versions without realizing it. Perhaps the default should be determined on whether the terminal is interactive?

- Is CODE_INSTALL_WITH_REPO set? Use that
- Is interactive? CODE_INSTALL_WITH_REPO = ask
- Is not interactive? CODE_INSTALL_WITH_REPO = write (old behavior)

The environment variable would give users control when it's important, we keep the old behavior when not interactive (automatically install repo for updates) and for interactive we ask users explicitly.

Then we can add to docs how `CODE_INSTALL_WITH_REPO` works and that you can run it like this:

```sh
CODE_INSTALL_WITH_REPO=0 ...
```

Maybe better to use strings instead of numbers so it's more obvious and you don't need to consult docs?

```sh
CODE_INSTALL_WITH_REPO="no" ...
CODE_INSTALL_WITH_REPO="ask" ...
CODE_INSTALL_WITH_REPO="write" ...
```

Also if `CODE_INSTALL_WITH_REPO` has an unexpected value we should probably warn and proceed with the default.

Reply:
There are two variables. I have changed WRITE_SOURCE, the internal variable, to use 'yes', 'ask', or 'no'.
Meanwhile, the if conditions still only check if CODE_INSTALL_WITH_REPO = true or if CODE_INSTALL_WITH_REPO = false.
I also added a new condition for non-interactive terminals later in the file.

Reply:
I have removed the environment variable to match with the docs, which currently only mention debconf.